File: PROGRAMMERS/LEVEL_02/Q12939.cpp
//https://programmers.co.kr/learn/courses/30/lessons/12939
//최댓값과 최솟값
#include <iostream>

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string s) {

    vector<int> v;

    for (int i = 0; i < s.size(); ++i)
    {
        int end = (s[i] == '-') ? i + 1 : i;
        
        // 숫자 변환
        int n = 0;
        while (end < s.size() && s[end] != ' ')
        {
            n *= 10;
            n += s[end] - '0';
            ++end;
        }
        // 음수 양수 반영
        n *= (s[i] == '-') ? -1 : 1;

        // vector push
        v.push_back(n);

        i = end;
    }

    // 최소값, 최대값 구하기
    auto minmax = minmax_element(v.begin(), v.end());

    return to_string(*minmax.first) + " " + to_string(*minmax.second);
}

int main()
{
    /*
    string s = "-1 -1";
    string s = "-1 -2 -3 -4	";
    string s = "1 2 3 4";
    */
    string s = "-192 0 -9 199 32432 -293848";

    cout << solution(s);

    return 0;
}

File: PROGRAMMERS/LEVEL_02/Q12939.java
// https://programmers.co.kr/learn/courses/30/lessons/12939
// 최댓값과 최솟값
package PROGRAMMERS.LEVEL_02;

public class Q12939 {

    public static void main(String[] args) {
        /*
        s	            return
        "1 2 3 4"	    "1 4"
        "-1 -2 -3 -4"	"-4 -1"
        "-1 -1"	        "-1 -1"
         */
        String s = "1 2 3 4";
        System.out.println(solution(s));
    }

    public static String solution(String s) {

        String[] numbers = s.split(" ");
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;

        for (String strNum : numbers) {
            int n = Integer.parseInt(strNum);
            max = Math.max(max, n);
            min = Math.min(min, n);
        }

        return min + " " + max;
    }
}
